.travis.yml

```diff
@@ -1,25 +1,55 @@
+sudo: required
 language: rust
-osx_image: xcode8.2
-
 rust:
   - stable
   - nightly
-
 os:
   - linux
   - osx
-
+osx_image: xcode8.2
+dist: trusty
 matrix:
   allow_failures:
     - rust: nightly
-
 cache:
   directories:
     - $HOME/.cargo
 
+before_install: |
+  if [ $TRAVIS_OS_NAME == "osx" ]; then
+    brew update
+    brew install python3
+    brew outdated pyenv || brew upgrade pyenv
+    brew install pyenv-virtualenv
+    eval "$(pyenv init -)"
+    eval "$(pyenv virtualenv-init -)"
+    sudo pyenv install 3.6.0
+    sudo pyenv virtualenv 3.6.0 general
+    sudo pyenv global general
+    sudo pip3 install -U setuptools
+
+  else
+    sudo curl -L https://raw.githubusercontent.com/pyenv/pyenv-installer/master/bin/pyenv-installer | bash
+    sudo apt-get update
+    export PATH="~/.pyenv/bin:$PATH"
+    eval "$(pyenv init -)"
+    eval "$(pyenv virtualenv-init -)"
+    sudo pyenv install 3.6.0
+    sudo pyenv virtualenv 3.6.0 general
+    sudo pyenv global general
+    sudo apt-get -y install python3-pip python-dev
```

Comment:
Why is this line needed here while on macOS it seems to be working without it?

Reply:
I ran a test with out it and it seems it is not required.

Reply:
@Melvie I guess you could use `push --force`

Reply:
@Melvie BTW my question was about the line `sudo apt-get -y install python3-pip python-dev` on Linux and not about pyenv and setuptools on macOS.

Reply:
I just saw that on line 561 of the travis output it says `sudo: pyenv: command not found` so it looks like the installation through pyenv did not work and `sudo apt-get -y install python3-pip python-dev` then installs python through the official Ubuntu repositories.

Reply:
@eyelash  Thanks for reviewing this and helping me out, really appreciate it. I added the pyenv and setup tools back to macOS and removed the sudo as you suggested from the Linux portion.